flytekit/core/python_auto_container.py

```diff
@@ -90,7 +93,23 @@ def __init__(
         kwargs["metadata"] = kwargs["metadata"] if "metadata" in kwargs else TaskMetadata()
         kwargs["metadata"].pod_template_name = pod_template_name
 
-        self._container_image = container_image
+        # Rename the `container_image` parameter to `image` for improved user experience.
+        # Currently, both `image` and `container_image` are supported to maintain backward compatibility.
+        # For more details, please refer to https://github.com/flyteorg/flyte/issues/6140.
+        if image is not None and container_image is not None:
+            raise ValueError(
+                "Cannot specify both image and container_image. "
+                "Please use image because container_image is deprecated and will be removed in the future."
+            )
+        elif container_image is not None:
+            warnings.warn(
```

Comment:
should we just make this info?  warning is a little too annoying I think.

Reply:
I feel like this is not annoying enough. `DeprecationWarning` does not really show up with default filter warnings. (`FutureWarning` would be more annoying)

If we really want users to change their code, then I think warnings should be a bit annoying. Because when we remove this feature, breaking code will be even more annoying.